stablehlo/dialect/TypeInference.cpp

```diff
@@ -3470,11 +3470,7 @@ LogicalResult verifyDynamicBroadcastInDimOp(
     std::optional<ArrayRef<int64_t>> knownNonexpandingDimensions,
     Value result) {
   auto operandType = operand.getType().dyn_cast<RankedTensorType>();
-  auto resultType = result.getType().dyn_cast<RankedTensorType>();
-
-  // If either the operand or result are unranked, there is very little
-  // to verify statically.
-  if (!operandType || !resultType) return success();
+  auto resultType = result.getType().cast<RankedTensorType>();
```

Comment:
This impl is likely shared between MHLO/StableHLO. For the duration of any parity gaps we'll need to make sure this logic makes its way into the MHLO op verify method.

Reply:
Ack. The impl is indeed shared. I have a patch ready.